dvc/main.py

```diff
@@ -64,6 +65,11 @@ def main(argv=None):
                 "unicode is not supported in DVC for Python 2 "
                 "(end-of-life January 1, 2020), please upgrade to Python 3"
             )
+        elif isinstance(exc, OSError) and exc.errno == errno.EMFILE:
+            logger.exception(
+                "too many open files error, for solution, please refer to our "
+                "user guide"
+            )
```

Comment:
It would look like "ERRROR: too many open files error..." which is odd. How about

> too many open files, please increase your `ulimit`. Refer to https://dvc.org/doc/user-guide/troubleshooting for more info.

This way there is a placeholder non-verbose solution and a referal to docs.

Reply:
Could skip `troubleshooting` part of the URL for now though. Not ideal, but at least we won't promise anything here 🙂

Reply:
Isn't `ulimit` unix-specific? That message would be amibguous for windows users

Reply:
@pared Yes, but it is generic enough that it brings the point across. More elaborate explanation will be in the linked doc. Your original message was similar but lacked the doc link.